Update architecture and storefront setup docs for ACCS and ACO (#746)
commerce-docs/microsite-commerce-storefront: src/content/docs/get-started/architecture.mdx

---
title: Storefront Architecture
description: Learn how to plan, develop, and launch a headless Adobe Commerce storefront with Edge Delivery Services. Understand blocks, drop-ins, boilerplate, and backend options.
---

import Aside from '@components/Aside.astro';
import Diagram from '@components/Diagram.astro';
import Link from '@components/Link.astro';

Headless Adobe Commerce storefronts on Edge Delivery Services use [drop-in components](/dropins/all/introduction/) that connect through APIs and integrate with the <Link href="https://github.com/hlxsites/aem-boilerplate-commerce" text="Commerce boilerplate" />.

Content is authored in documents (DA.live, Google Docs, SharePoint), in-context (Universal Editor, Storefront Builder), or AEM Sites—and rendered as blocks. See the <Link href="https://www.aem.live/docs/authoring-guide" text="AEM authoring guide" /> for details.

This page explains the architecture, then points you to [Backend options](/get-started/backends/) to identify your backend and confirm prerequisites.

## Architecture overview

<Diagram type="mermaid" code={`

> graph TB

Reviewer: @fnhipster do you have a better diagram? I don't know how I feel about this one 😆

    EDS["<b>Edge Delivery Services</b><br/>Hosting & CDN"]
    Blocks["<b>Blocks</b><br/>Author in DA.live"]
    CommerceBlocks["<b>Commerce Blocks</b><br/>Author in DA.live"]
    EDS --> Blocks
    EDS --> CommerceBlocks

>     Blocks --> Boilerplate["<b>Commerce Boilerplate</b><br/>config.json + API clients"]

Reviewer: No config.json please

---
title: Storefront Architecture
description: Learn how to plan, develop, and launch a headless Adobe Commerce storefront with Edge Delivery Services.
---

import Aside from '@components/Aside.astro';
import { CardGrid, Card, Tabs, TabItem, Steps } from '@astrojs/starlight/components';
import Callouts from '@components/Callouts.astro';
import Diagram from '@components/Diagram.astro';
import Task from '@components/Task.astro';
import Tasks from '@components/Tasks.astro';
import Options from '@components/Options.astro';
import Option from '@components/Option.astro';
import Link from '@components/Link.astro';
import TableWrapper from '@components/TableWrapper.astro';

Headless Adobe Commerce storefronts on Edge Delivery Services use [drop-in components](/dropins/all/introduction/) that connect through APIs. Drop-ins integrate with the <Link href="https://github.com/hlxsites/aem-boilerplate-commerce" text="Commerce boilerplate" />. Content is authored in documents (DA.live, Google Docs, Word) and rendered as blocks. This page helps you identify your backend and confirm prerequisites before setup.

## Backend options

> Storefront supports three backend types. Follow only the guidance for your backend. Prerequisites and setup differ, and mixing steps from different backends causes errors.

Reviewer 2: ```suggestion
Storefront supports three backend types. Follow only the guidance for the backend that you're integrating with your storefront. Each backend has different prerequisites and setup steps, and mixing instructions across backends causes errors.
```

> - **Commerce PaaS** — Existing PaaS customer without Adobe Commerce as a Cloud Service or Adobe Commerce Optimizer licenses

Reviewer 2: Avoid "existing" or "new" because it doesn't matter.

```suggestion
- **Commerce PaaS** — Adobe Commerce on cloud or on-premises projects
```

> - **Adobe Commerce as a Cloud Service** — New ACCS customer; license includes Storefront

Reviewer 2: Do  all ACCS customers have a storefront license?   Not sure what the license is called exactly. Ask Fluffy.
```suggestion
- **Adobe Commerce as a Cloud Service** — Customers with an Adobe Commerce as a Cloud Service, which includes a license for the Commerce Storefront powered by Edge Delivery Services.
```

> - **Adobe Commerce Optimizer** — New Optimizer customer, or existing PaaS customer who added Optimizer

Reviewer 2: ```suggestion
- **Adobe Commerce Optimizer** or **Adobe Commerce Optimizer Connector**  —  The connector enables customers with Adobe Commerce PaaS deployments to export data to Adobe Commerce Optimizer and create Commerce Storefronts on Edge Delivery Services.
```

<TableWrapper nowrap={[0]}>

> | Feature | Commerce PaaS | Adobe Commerce as a Cloud Service | Adobe Commerce Optimizer |

Reviewer 2: ```suggestion
| Feature | Commerce PaaS | Adobe Commerce as a Cloud Service | Adobe Commerce Optimizer <br>Adobe Commerce Optimizer Connector |
```

|---------|---------------|-----------------------------------|--------------------------|
| **License includes Storefront** | No | Yes | Yes |

> | **Setup** | Manual install required | Fully automated | Fully automated |

Reviewer 2: Not sure how to add it to the table, but the Connector requires manually installing the PHP extension on the PaaS platform and configuring the ACO UI.  I will provide a link to the documentation for the connector in the next day or two.

| **Version** | v2.4.7+ self-managed | Auto-managed | Fully managed SaaS |

</TableWrapper>

<Options label="Prerequisites" numbered={false}>
<Option>

### Commerce Platform-as-a-Service (PaaS)

> - **Product license**: Adobe Commerce Cloud or on-premises (Magento Open Source not supported)

Reviewer 2: another option:  Adobe Commerce cloud or on-premises with the Adobe Commerce Optimizer connector;  this extension requires the storefront compatibility package. ([PR](https://experienceleague-review.corp.adobe.com/docs/commerce/aco-optimizer-connector/overview.html))

- **Version**: v2.4.7 or later
- **Storefront Compatibility package**: Required. See [Storefront Compatibility package](/setup/configuration/storefront-compatibility/install/) (2.4.7 or 2.4.8).

> - **Storefront services** (manual install): Data Connection, Services Connector, Catalog Service, Live Search, Product Recommendations. See <Link href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/integration-services/saas" text="Commerce Services Connector" />.

Reviewer 2: so to clarify for the PaaS + ACO connector backend -- the manual install list is "Adobe Commerce Optimizer Connector"  -- Not sure if the Data Connection is supported when using the ACO Connector.

:::note

> **Adobe Commerce Optimizer integration:** PaaS + Adobe Commerce Optimizer is available. See [Luma Bridge](/setup/discovery/luma-bridge/) for migration.

Reviewer 2: so if you have PaaS and you have a LUMA store, you use Luma Bridge to migrate if you install the ACO Connector.  Or can PaaS customers use Luma bridge without having ACO license?

:::

</Option>

<Option>

### Adobe Commerce as a Cloud Service

> - **Product license**: Adobe Commerce as a Cloud Service (License includes Storefront)

Reviewer 2: oh -- I see now --  License includes storefront -- not "if license includes storefront"
